analyzers/tests/SonarAnalyzer.TestFramework.Test/Common/SourceGeneratorProviderTest.cs

```diff
@@ -46,11 +47,18 @@ public class SourceGeneratorProviderTest
     [TestMethod]
     public void LatestSdkFolder_ReturnsCorrectPath()
     {
-        var expectedPath = Directory.GetDirectories(Path.Combine(Directory.GetParent(typeof(object).Assembly.Location).Parent.Parent.Parent.FullName, "sdk"), $"{typeof(object).Assembly.GetName().Version.Major}.*", SearchOption.TopDirectoryOnly)
-                                    .OrderByDescending(x => new DirectoryInfo(x).Name)
-                                    .FirstOrDefault();
-        SourceGeneratorProvider.LatestSdkFolder().Should().Be(expectedPath);
+        var latestSdkFolder = SourceGeneratorProvider.LatestSdkFolder();
+        var latestSdkVersion = Version.Parse(Path.GetFileName(latestSdkFolder));
+
+        var parentDirectory = Directory.GetParent(latestSdkFolder);
+        parentDirectory.Name.Should().Be("sdk", "Parent directory of the latest SDK should be 'sdk'");
+
+        Directory.GetDirectories(parentDirectory.FullName, $"{typeof(object).Assembly.GetName().Version.Major}.*", SearchOption.TopDirectoryOnly)
```

Comment:
Add a check that confirms that `typeof(object).Assembly.GetName().Version.Major` is the same as `latestSdkVersion.Major`.